homeassistant/components/lock/__init__.py

```diff
@@ -156,6 +156,22 @@ async def async_open(self, **kwargs: Any) -> None:
         """Open the door latch."""
         await self.hass.async_add_executor_job(ft.partial(self.open, **kwargs))
 
+    def setpin(self, **kwargs: Any) -> None:
+        """Set Pin on the device."""
+        raise NotImplementedError()
+
+    async def async_setpin(self, **kwargs: Any) -> None:
+        """Set Pin on the device."""
+        await self.hass.async_add_executor_job(ft.partial(self.setpin, **kwargs))
+
+    def clearpin(self, **kwargs: Any) -> None:
+        """Clear Pin on the device."""
+        raise NotImplementedError()
+
+    async def async_clearpin(self, **kwargs: Any) -> None:
+        """Clear Pin on the device."""
+        await self.hass.async_add_executor_job(ft.partial(self.clearpin, **kwargs))
+
```

Comment:
We won't add new services to base entities unless approved in an architecture discussion.
If that has been done, please add a link to it.
If not, either open one or expose these features as service in the vera lock platform.

Reply:
components/vera/lock.py code has also been update to support this change